### PR TITLE
Upgrade to Python 3.11.x

### PR DESCRIPTION
**Pull request by robhudson**

### CODE REVIEW

File: .github/workflows/pull_request_tests.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions/setup-python@v4
         with:
-          python-version: "3.9"  # matches current Python in production
+          python-version: "3.11"  # matches current Python in production
```
Comment (stevejalim) [inline .github/workflows/pull_request_tests.yml:35]: **thought:** For Nucleus we went straight to 3.12. I wonder if that's OK here, too

File: bedrock/security/management/commands/update_security_advisories.py [hunk 2]
```diff
@@ -65,7 +65,7 @@ def add_or_update_advisory(data, html):
     :return: SecurityAdvisory
     """
     mfsa_id = data.pop("mfsa_id")
-    year, order = [int(x) for x in mfsa_id.split("-")]
+    year, order = (int(x) for x in mfsa_id.split("-"))
```
Comment (stevejalim) [inline bedrock/security/management/commands/update_security_advisories.py:68]: **question (non-blocking):** Is there much to be gained by making this a generator, given we immediately exhaust it?

File: docs/install.rst [hunk 3]
```diff
@@ -121,7 +121,7 @@ The following assumes you are on MacOS, using ``zsh`` as your shell and `Homebre
 as your package manager. If you are not, there are installation instructions for a variety of
 platforms and shells in the READMEs for the two pyenv projects.
 
-**Install Python 3.9.10 with pyenv**
+**Install Python 3.11.x with pyenv**
```
Comment (stevejalim) [inline docs/install.rst:124]: **praise (non-blocking):** Thank you for updating the docs too!

**Review by stevejalim — APPROVED** [review]
Nice! r+wc

📺

File: .github/workflows/pull_request_tests.yml (hunk 1 above)
Comment (robhudson) [reply .github/workflows/pull_request_tests.yml:35]: Yeah I thought about that. But 3.12 is only on the 2nd patch release. I was thinking we would want bedrock to be a little more careful than the others.

File: bedrock/security/management/commands/update_security_advisories.py (hunk 2 above)
Comment (robhudson) [reply bedrock/security/management/commands/update_security_advisories.py:68]: pyupgrade did it so I wasn't going to undo it. My feeling was to leave it as it's probably generally better and it's a good pattern to follow? Open to other opinions though.

**Pull request merged**